src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/SwitchStatementHighlighter.cs

```diff
@@ -26,12 +27,14 @@ internal class SwitchStatementHighlighter : AbstractKeywordHighlighter<SwitchSta
                 {
                     spans.Add(label.Keyword.Span);
                     spans.Add(EmptySpan(label.ColonToken.Span.End));
-                }
 
-                foreach (var statement in switchSection.Statements)
-                {
-                    HighlightRelatedKeywords(statement, spans);
+                    if (label is CasePatternSwitchLabelSyntax patternLabel && patternLabel.WhenClause != null)
+                    {
+                        spans.Add(patternLabel.WhenClause.WhenKeyword.Span);
```

Comment:
not sure highlighting these is what we want.

Reply:
i noticed that `when` is highlighted on catch clauses and though it would be appropriate for consistency to have that here as well... but i can remove it

Reply:
Gotcha.  I don't think it's a good idea for either TBH :)   THe purpose of this highlighting is to quickly help jump between the relation between  constructs that can be far apart.  i.e. "this case belongs to thsi switch" (and vice versa).  i don't see the same need for 'when' given that it's much more tightly associated with a 'case' clause making it obvious what the relation is.

Reply:
so should i just remove it?

Reply:
My personal preference would be that it was not added as i don't think it adds value, or fits into the goals of the keyword highlighting feature.  You can ping @dotnet/roslyn-ide  for more minds on this if you feel strongly like this should stay.

Reply:
Note that "jump between constructs far apart" is already somewhat broken: we would highlight breaks inside of a case, which means you might have to an arbitrary number of jumps to get around. I think this is fine, especially because we already do it in other places.

Reply:
removed